src/main/java/com/powsybl/balances_adjustment/util/CountryArea.java

```diff
@@ -6,61 +6,115 @@
  */
 package com.powsybl.balances_adjustment.util;
 
-import com.powsybl.iidm.network.Country;
-import com.powsybl.iidm.network.Network;
-import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.iidm.network.*;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
 
 /**
  * @author Ameni Walha {@literal <ameni.walha at rte-france.com>}
+ * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
  */
-public class CountryArea extends AbstractNetworkArea {
+public class CountryArea implements NetworkArea {
 
-    private final Country country;
+    private final List<Country> countries;
 
-    public CountryArea(Country country) {
-        this.country = country;
+    private List<DanglingLine> danglingLineBordersCache;
+    private List<Line> lineBordersCache;
+    private List<HvdcLine> hvdcLineBordersCache;
+
+    public CountryArea(Country... countries) {
+        this.countries = Arrays.asList(countries);
+        resetCache();
     }
 
-    public Country getCountry() {
-        return country;
+    public List<Country> getCountries() {
+        return countries;
     }
 
     @Override
-    public List<VoltageLevel> getAreaVoltageLevels(Network network) {
-        return network.getVoltageLevelStream()
-                .filter(voltageLevel -> voltageLevel.getSubstation().getCountry().isPresent())
-                .filter(voltageLevel -> voltageLevel.getSubstation().getCountry().get().equals(country))
-                .collect(Collectors.toList());
+    public double getNetPosition(Network network) {
+        cacheAreaBorders(network);
```

Comment:
There is something strange in the design: how do you now that the cache is up-to-date? The network passed in parameters could be different from the previous call.
Should the network be an attribute of this class?